src/Files.App/ViewModels/Properties/HashesViewModel.cs

```diff
@@ -41,76 +28,89 @@ public HashInfoItem SelectedItem
 
 		public ObservableCollection<HashInfoItem> Hashes { get; set; }
 
-		private Stream _stream;
+		private ListedItem _item;
 
-		public AsyncRelayCommand LoadAndCalcHashesCommand { get; set; }
+		private CancellationTokenSource _cancellationTokenSource;
 
-		public CancellationTokenSource CancellationTokenSource { get; set; }
+		private Dictionary<string, bool> _showHashesDictionary;
 
-		private bool _isLoading;
-		public bool IsLoading
-		{
-			get => _isLoading;
-			set => SetProperty(ref _isLoading, value);
-		}
-
-		public async Task ExecuteLoadAndCalcHashesCommandAsync(CancellationToken cancellationToken)
+		public HashesViewModel(ListedItem item)
 		{
-			try
-			{
-				IsLoading = true;
+			_item = item;
+			_cancellationTokenSource = new();
+			_showHashesDictionary = UserSettingsService.PreferencesSettingsService.ShowHashesDictionary;
 
-				_stream = File.OpenRead(Item.ItemPath);
-
-				CanAccessFile = true;
-				await GetHashesAsync(cancellationToken);
-			}
-			catch (OperationCanceledException)
-			{
-				CanAccessFile = false;
-			}
-			catch (Exception)
+			Hashes = new()
 			{
-				CanAccessFile = false;
-			}
-			finally
+				new() { Algorithm = "MD5" },
+				new() { Algorithm = "SHA1" },
+				new() { Algorithm = "SHA256" },
+				new() { Algorithm = "SHA384" },
+				new() { Algorithm = "SHA512" },
+			};
+			Hashes.ForEach(x =>
 			{
-				IsLoading = false;
-			}
+				x.PropertyChanged += HashInfoItem_PropertyChanged;
+				if (_showHashesDictionary.TryGetValue(x.Algorithm, out var value)) {
+					x.IsEnabled = value;
+				} 
+				else
+				{
+					x.IsEnabled = x.Algorithm switch
+					{
+						"MD5" => true,
+						"SHA1" => true,
+						"SHA256" => true,
+						"SHA384" => false,
+						"SHA512" => false,
+						_ => throw new InvalidOperationException()
```

Comment:
Should not occur an exception. Please set default like MD5

Reply:
The default value should be explicitly specified for each algorithm. So I think it should be leave as it is in case someone forgets to set the default value when adding an algorithm.

Reply:
Should not break the app.